src/callbacks_step/analysis_dg2d_parallel.jl

```diff
@@ -12,31 +12,92 @@ function calc_error_norms(func, u, t, analyzer,
   # TODO: MPI, we should improve this; maybe we should dispatch on `u`
   #       and create some MPI array type, overloading broadcasting and mapreduce etc.
   #       Then, this specific array type should also work well with DiffEq etc.
-  l2_error, linf_error = invoke(calc_error_norms,
-    Tuple{typeof(func), typeof(u), typeof(t), typeof(analyzer), TreeMesh{2},
-          typeof(equations), typeof(initial_condition), typeof(dg), typeof(cache),
-          typeof(cache_analysis)},
-    func, u, t, analyzer, mesh, equations, initial_condition, dg, cache, cache_analysis)
-
-  # Since the local L2 norm is already normalized and square-rooted, we need to undo this first
-  total_volume = mesh.tree.length_level_0^ndims(mesh)
-  global_l2_error = Vector(l2_error.^2 .* total_volume)
-  global_linf_error = Vector(linf_error)
-  MPI.Reduce!(global_l2_error, +, mpi_root(), mpi_comm())
-  MPI.Reduce!(global_linf_error, max, mpi_root(), mpi_comm())
+  l2_errors, linf_errors = calc_error_norms_per_element(func, u, t, analyzer,
+                                                        mesh, equations, initial_condition,
+                                                        dg, cache, cache_analysis)
+
+  nvars = length(l2_errors[1])
+  T = eltype(l2_errors)
+
+  # Convert errors from Vector of SVectors to Matrix for MPI communication
+  FT = eltype(l2_errors[1])
+  l2_errors = reshape(reinterpret(FT, l2_errors), nvars, :) |> Matrix
+  linf_errors = reshape(reinterpret(FT, linf_errors), nvars, :) |> Matrix
```

Comment:
Why do you need a `Matrix`? Can't you communicate a `Vector` of `SVector`s? Can't you communicate a `reinterpret`ed array? Please, try to write comments that explain *why* you are doing something, not *what* you are doing - the latter should be clear from the code anyway

Reply:
I couldn't recall the exact error I got when I initially tried to communicate a `Vector` of `SVector`s so I just revisited this and could make communicating a `Vector` of `SVector`s work (see 3c0e921). Apparently I made a mistake somewhere else back then.